### PR TITLE
If no unit is given, no '[]' is printed.

### PR DESCRIPTION
**Pull request by reimersa**

### CODE REVIEW

File: hepdata_lib/__init__.py [hunk 1]
```diff
@@ -134,7 +134,10 @@ def make_dict(self):
         Except for debugging purposes, no user should have to call this function.
         """
         tmp = {}
-        tmp["header"] = {"name": self.name, "units": self.units}
+        if self.units != '':
```
Comment (AndreasAlbert) [inline hepdata_lib/__init__.py]:
```suggestion
        tmp["header"] = {"name": self.name}
        if self.units:
            tmp["header"]["units"] = self.units
```

File: hepdata_lib/__init__.py [hunk 2]
```diff
@@ -134,7 +134,10 @@ def make_dict(self):
...
+            tmp["header"] = {"name": self.name, "units": self.units}
```
Comment (AndreasAlbert) [inline hepdata_lib/__init__.py]:
```suggestion
            
```

File: hepdata_lib/__init__.py [hunk 3]
```diff
@@ -134,7 +134,10 @@ def make_dict(self):
...
+        else:
```
Comment (AndreasAlbert) [inline hepdata_lib/__init__.py]:
```suggestion

```

File: hepdata_lib/__init__.py [hunk 4]
```diff
@@ -134,7 +134,10 @@ def make_dict(self):
...
+            tmp["header"] = {"name": self.name}
```
Comment (AndreasAlbert) [inline hepdata_lib/__init__.py]:
```suggestion

```

**Comment by AndreasAlbert** [comment]
Hi @reimersa, thanks for this contribution! The empty "[]" thing is indeed quite annoying. I would propose to slightly change your suggested code to avoid the duplication of the assignment line. I am not sure whether this suggestion will show up correctly with the in-code github comment, so here it is again in plain text:

```
        tmp["header"] = {"name": self.name}
        if self.units:
            tmp["header"]["units"] = self.units
```

Let me know what you think.

**Comment by reimersa** [comment]
Hi @AndreasAlbert ,

sure, that makes perfect sense to me, thanks for the suggestion!

**Pull request merged**